### PR TITLE
fix: position (character & line) to be extracted in any order

### PR DESCRIPTION
**Pull request by GoNZooo**
I noticed that I was getting a `badmatch` error because `<<"character">>` & `<<"line">>` came in out of order, so I made a function to handle either ordering.

### CODE REVIEW

File: apps/lfe-ls/src/lsp-proc.lfe [hunk 1]
```diff
@@ -212,6 +212,12 @@ Handler functions (like `%on-initialize-req`) are expected to return:
            ,state
            null)))))
 
+(defun extract-position-data
```
Comment (mdbergmann) [inline apps/lfe-ls/src/lsp-proc.lfe]: Excellent.
Can you please prefix a "%" character to indicate internal function.

**Review by mdbergmann — CHANGES_REQUESTED** [review]

**Comment by mdbergmann** [comment]
I was thinking about adding a test case in lsp-proc-tests.
There are two or three completion tests, maybe you can just turn around the order of line and character in one of them to have this covered.

File: apps/lfe-ls/src/lsp-proc.lfe (hunk 1 above)
Comment (GoNZooo) [reply apps/lfe-ls/src/lsp-proc.lfe]: Done in a906510db57a786389b801c241bd4c98162bfdc2

**Comment by GoNZooo** [comment]
> I was thinking about adding a test case in lsp-proc-tests. There are two or three completion tests, maybe you can just turn around the order of line and character in one of them to have this covered.

I've added two test cases now in   33dd82cc437cba3bcfeeb6b3f8d33aaeacc67f14, they are just variations of other test cases where the order is swapped, as you suggested.

File: apps/lfe-ls/src/lsp-proc.lfe [hunk 2]
```diff
@@ -270,7 +276,7 @@ Handler functions (like `%on-initialize-req`) are expected to return:
                        (#(#"resolveProvider" false)
                         #(#"triggerCharacters" (#"(" #":" #"'"))))
                      #(#"textDocumentSync"
-                       (#(#"openClose" true) #(#"change" 1)))
+                       (#(#"openClose" true) #(#"change" 1) #(#"save" (#(#"includeText" false)))))
```
Comment (GoNZooo) [inline apps/lfe-ls/src/lsp-proc.lfe:279]: I've  also added this in order to have `didSave` events be sent by the client. It's possible that some clients assume they should, but in order to get Neovim LSP to send them I had to say up front that this was something supported by the server.

Comment (mdbergmann) [reply apps/lfe-ls/src/lsp-proc.lfe:279]: This is great. But please don't package too much into a PR. Different topics require different PRs to keep visibility.
A PR title "fix: position (character & line) to be extracted in any order" should only do that, but not more.

So it's it's not too much trouble, please create a separate PR. It should not depend on the open one.

Comment (mdbergmann) [reply apps/lfe-ls/src/lsp-proc.lfe:279]: Anyway, it's OK first first PR.

**Review by mdbergmann — APPROVED** [review]

**Pull request merged**